tests/mbed/enc/CMakeLists.txt

```diff
@@ -22,9 +22,23 @@ function(add_mbed_test_enclave NAME)
 
   set(MBEDTLS_TESTS_DIR "${PROJECT_SOURCE_DIR}/3rdparty/mbedtls/mbedtls/tests")
 
+# Use Perl that is found in the Path
+# If Perl is not found on Windows, use Perl from the Git/usr/bin directory
+# and report error on Linux
+  if (WIN32)
+    find_program(GIT git)
+    get_filename_component(GIT_DIR ${GIT} DIRECTORY)
+    set(PERL_HINTS "C:/Program Files/Git/usr/bin" "${GIT_DIR}/../usr/bin")
+  endif ()
+
+  find_program(PERL perl HINTS ${PERL_HINTS})
+  if (NOT PERL)
+    message(FATAL_ERROR "Perl not found! Cannot build mbedtls Test Suites without Perl!")
+  endif ()
+
   add_custom_command(
     OUTPUT test_suite_${NAME}.c
-    COMMAND ${PERL_EXECUTABLE} ${MBEDTLS_TESTS_DIR}/scripts/generate_code.pl ${MBEDTLS_TESTS_DIR}/suites test_suite_${suite_name} test_suite_${data_name}
+    COMMAND ${PERL} ${MBEDTLS_TESTS_DIR}/scripts/generate_code.pl ${MBEDTLS_TESTS_DIR}/suites test_suite_${suite_name} test_suite_${data_name}
```

Comment:
It's needed on Windows, and it's still necessary to _have_ perl despite the shebang, so we might as well share the logic and check for it. That gives us much better error handling.

Reply:
(Perl is not necessarily in the PATH on Windows, hence the use of PERL_HINTS above to teach CMake where else to look for Perl. It simplifies things for the user.)

Reply:
@andschwa I just verified that git-bash correctly processes shebang. Otherwise git will be unusable on windows.

Reply:
@anakrish - The original code invoked generate_code.pl with the Perl executable.
>>Can you check if instead of ${PERL}, you can use ${BASH} ==> No, this will not work on Windows as Git bash  is in c:\program files\Git\bin and perl.exe is in c:\program files\Git\usr\bin. Users don't automatically get Perl.exe with Git unless the c:\program files\Git\usr\bin directory was added to the path during the Git installation.

Reply:
Well, I see what Anand is saying. If we _explicitly_ use Git Bash (not WSL Bash) then we'd be guaranteed to have Perl in the path and the shebang will work.

But that _requires_ Git Bash, and I know we already do that elsewhere, but I'd rather not do it again. As-is the code will work does not depend on Git Bash, just Perl.

Reply:
Yep, I concur.